### ISSUE TITLE
Add a 'force upload/sync' button to automatic uploads

### ISSUE THREAD

**Issue by ppattard** [nextcloud/android#7019]
### Is your feature request related to a problem? Please describe.
When putting Android into powersave mode (with network disabled in background), automatic uploads won't start when a new picture is taken from the camera. I always have to exit the powersave mode first, go into Nextcloud and wait for it to find the new pictures. It usually takes between 30s and up to several minutes for it to actually start the upload.

### Describe the solution you'd like
A force upload button in the UI, for instance on uploads history panel, or from the panel where you configure upload folders, to trigger the new file detection and automatic upload would be a nice workaround to avoid having to wait.

### Describe alternatives you've considered
Fixing the upload that do not start when in powersave mode (even when the screen is on and not locked), but I'm not sure if this is possible or if this is a limitation of Android.

### Additional context
--

**Comment by Derkades** [comment]
I agree that a force upload button would be useful. In my case I have it configured to only upload when charging, but there are times where I need a file uploaded immediately.

> Fixing the upload that do not start when in powersave mode (even when the screen is on and not locked), but I'm not sure if this is possible or if this is a limitation of Android.

I don't think this is a good solution. Apps should save power when in power saving mode. If a user enables power saving mode in some situation where they need their phone to keep working and are not able to charge it, they usually wouldn't want nextcloud to waste this energy uploading files.

A force button will make everyone happy.

**Comment by datenheim** [comment]
This would be useful indeed.
Possibly can be put together with "Pause/restart aborted auto-uploads?" #6823

**Comment by tobiasKaminsky** [comment]
Force upload is already implement: if you go to upload view and swipe to refresh, then a manual auto upload detection job is triggered.

**Comment by Derkades** [comment]
It may detect new files but it does not "force upload" them. If it is waiting for battery saver to be turned off or the phone to start charging it will still be waiting.

**Comment by tobiasKaminsky** [comment]
Ah right :+1: 
Then I would suggest a "play/resume/restart" button next to "cancel" on active group, which then overrides all checks as a one-timer.

**Comment by codereptile** [comment]
Any progress on this? I would LOVE to see this, because I often find it difficult to start uploading existing files when creating a new 'auto upload' catalogue

**Comment by panasenco** [comment]
Ran into the same issue, my workaround:

### Use custom syncs instead of predefined ones
1. Go to Settings > Auto upload
1. Disable syncs for all predefined places there.
2. Instead, hit the three dots in the top right and click "Set up a custom folder". Select the local and remote folders. Be sure to set "What to do if the file already exists?" to **Skip uploading**.
3. Save the sync. It should start.

### Delete and re-create the custom sync to force sync
When you need to force sync:
1. Go to Settings > Auto upload.
2.  Click the three dots next to the custom sync > Configure.
3. **Delete** the custom sync.
4. Recreate as described above.
5. The sync should start. All files that already exist in NextCloud will be quickly skipped.

**Comment by regs01** [comment]
This is indeed very sharp issue. Quite often you make a photo and screenshot and need it immediately on you computer, but you have to wait half a hour until it syncs automatically.

There is a pause button in Upload status screen. Just make it force resync, every time you resuming syncing, so you can just press pause and resume to force do now.

**Comment by shalak** [comment]
Is there a workaround simpler than what was already suggested? It doesn't start when I plug in the phone to charger...

**Comment by luxzg** [comment]
> Is there a workaround simpler than what was already suggested? It doesn't start when I plug in the phone to charger...

My main use for Nextcloud app was to sync photos, but the issues with auto-upload made me look for alternatives.

While it won't fix this app, swapping to simpler app that actually does the sync when and how I want *is* a simpler solution vs waiting years to get a single button.

I've been using Zpush app on Android for almost 2 years now, on 4th device now, no bells, just a pure simple sync. It doesn't waste much energy at all, and has a simple way to press a button and force an immediate sync.

If you want you can check it here:
https://play.google.com/store/apps/details?id=de.maza.zpush

**Comment by luxzg** [comment]
> > While it won't fix this app, swapping to simpler app that actually does the sync when and how I want _is_ a simpler solution vs waiting years to get a single button.
> 
> Unfortunately, ZPush is not FOSS and as such isn't really a suitable replacement here.

It's free, no advertisments, respective of privacy, and that its single-dev doesn't share code doesn't bother me really, as the person is nice in communication, offers his work for free to community, and fixed issues I noticed and made improvements suggested in a matter of days. While at the same time some FOSS alternatives refuse to do that for 4+ years. In my book FOSS doesn't win every time just because it's FOSS. But to each their own, I gain nor lose anything by use of one or the other. I still have Nextcloud app installed on my primary phone "just in case", even though all the work for almost 2 years had been handled by proposed alternative. I would really wish for Nextcloud app to fulfill everyone's needs, but that is never a reality of app development.

**Comment by toddpfaff** [comment]
It still seems that there is no way to force an "Instant Upload" of new photos in the Android NextCloud app.  One would think that an Instant Upload would be somewhat Instant but it's not.  And there still seems to be no way to force this to happen (e.g. click pause and then resume doesn't help). The upload will happen eventually but there doesn't appear to be any rhyme or reason as to when it happens.  I have my Instant Upload settings such that it will upload anywhere, anytime (i.e. it doesn't need to be plugged in for charging, it can be on any network, wifi or cell data, so it's none of those things that's delaying uploads).  Is the Instant Upload feature just on some long timer or something that like?  That is, does it only attempt an Instant Upload once every 15 minutes or some interval that the user has no control of?  This is inconvenient when one is trying to get a workflow done quickly - take photo, expect NextCloud client to Instant Upload it somewhat Instantly, then work with the uploaded file from another location.  After all these years, let's please see Instant Upload be Instant, or able to be forced via the pause/resume button for example.  Or rename the feature to Periodic-Upload-Where-Period-is-X-Minutes-That-The-User-Can-Not-Control.

And, yes, I realize I can just manually upload the file.  That's what I do now when the Instant Upload isn't Instant.

I'm going to start reading the code myself to try to determine what the current algorithm is, what the timer may be, etc.

**Comment by codereptile** [comment]
Off-topic, but helpful: I've switched to using the "**foldersync**" app by tacit dynamics. It features robust scheduled and 'on change' syncing, as well as the aforementioned sync now button. 

P.S. I understand that this is not a solution, but some might find this workaround useful, as this is a rather popular issue.

**Comment by pciavald** [comment]
quitting (not backgrounding) and reopening the app made it refresh picture for me just now looking for an answer, maybe this will help others ?

**Comment by regs01** [comment]
No, it does not

**Comment by shalak** [comment]
Is it possible to put a bounty on this repo? Lack of this feature is frequent source of frustration for me, I'm willing to donate to get it done.

**Comment by asavageiv** [comment]
> This is indeed very sharp issue. Quite often you make a photo and screenshot and need it immediately on you computer, but you have to wait half a hour until it syncs automatically.

A workaround for this use case is to go to the photo in the Files app do click Share > Nextcloud > Select upload folder configured in Nextcloud > Upload

**Comment by SK-Crow** [comment]
+1 this is a major issue I'm running into since Android changed a while back.  I end up emailing myself photos or manually uploading them.  I used to be able to force it by killing the app process and re-opening, but that no longer works.

**Comment by Mario-Hofstaetter** [comment]
Although I had energy saver mode enabled since a hike yesterday, some files where uploaded (from camera) nevertheless.

However, **out of 129 pictures and videos taken only yesterday, 24 (!!) where missing** in nextcloud, the app acting like it does not know about them. 
Disabling energy saver does nothing. Clearly as issues like this one here are open for years, nextcloud does not have the capacity to fix these issues, and I as a private person don't have the money to buy "enterprise support" from nextcloud.

To any reader of this issue, I must recommend not using the camera auto upload feature of the app, as it is completely broken and unreliable, and dozens of issues and topics online document that for years.

I am back to syncing with USB cable on the PC (wow) .. might fallback to onedrive for autoupload (and then use rclone to copy onedrive -> nextcloud), which defeats the purpose of switching to nextcloud.

**Comment by tobiasKaminsky** [comment]
Idea: Inside upload view --> "battery optimization header" -> add second option to "sync now"

**Comment by regs01** [comment]
> Idea: Inside upload view --> "battery optimization header" -> add second option to "sync now"

It needs to be two different buttons sync now for sync and force upload for autoupload. Second would be better in status screen next to pause button.

**Comment by rain1** [comment]
Is there any chance to get it implemented?

**Comment by alperozturk96** [comment]
Hello,

PR is created for this feature: https://github.com/nextcloud/android/pull/17396.

You can try last generated QA APK and share your feedback with us.

About PR: 

App's shows a banner if device is in power save mode and this affects the auto upload behaviour such as delays.

This PR adds "Sync now" button inside the banner to re-trigger auto upload via overriding the battery optimization.

**Comment by regs01** [comment]
is it autoupload now or sync now? those just two different things.

**Comment by chesamo** [comment]
#17396 doesn't address this issue directly per comment [5175294367](/nextcloud/android/pull/17396#issuecomment-5175294367):

> &hellip;this button only re-trigger auto upload via bypassing the battery optimization. It does not ignore every check and try to force upload every file.

If the "fix" is only accessible by entering battery saving mode, that isn't a fix.

**Comment by regs01** [comment]
@alperozturk96 
Yeah, that PR is not related to this issue. It's to bypass battery optimizations, when auto triggered. This issue is about let users autoupload at demand.

**Comment by rain1** [comment]
Basically NC is doing everything they can to avoid doing what users actually ask.
